tsdb/head_append.go

```diff
@@ -576,10 +624,16 @@ func (a *headAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels
 		return 0, storage.ErrNativeHistogramsDisabled
 	}
 
-	if t < a.minValidTime {
+	// For OOO inserts, this restriction is irrelevant and will be checked later once we confirm the histogram sample is an in-order append.
```

Comment:
I find this comment a bit confusing, would leave it at "Fail fast if ooo is disabled". Then the appendable check would do the rest.

Reply:
updated